libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -77,9 +77,24 @@ string IRGenerationContext::functionName(VariableDeclaration const& _varDecl)
 	return "getter_fun_" + _varDecl.name() + "_" + to_string(_varDecl.id());
 }
 
-string IRGenerationContext::virtualFunctionName(FunctionDefinition const& _functionDeclaration)
+string IRGenerationContext::enqueueFunctionForCodeGeneration(FunctionDefinition const& _function)
 {
-	return functionName(_functionDeclaration.resolveVirtual(mostDerivedContract()));
+	m_functionGenerationQueue.insert(&_function);
+	return functionName(_function);
+}
+
+string IRGenerationContext::enqueueVirtualFunctionForCodeGeneration(FunctionDefinition const& _function)
```

Comment:
Maybe it would be better to not have a special function for this?

Reply:
Function removed.